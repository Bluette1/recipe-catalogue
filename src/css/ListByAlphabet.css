h2 {
  text-align: center;
  font-family: 'Mountains of Christmas', cursive;
  color: #2d2013;
}

.list-by-alphabet {
  background-color: #d57d1f;
  margin: 10%;
  border: 3px solid #2d2013;
  padding: 5%;
}

.list-by-alphabet .ul {
  list-style: none;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}

.ul li {
  padding: 8px;
  font-family: 'Elsie', cursive;
}

a {
  text-decoration: underline;
  color: #008000;
}
