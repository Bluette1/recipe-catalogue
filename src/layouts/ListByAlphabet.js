import React from 'react';

const ListByAlphapet = () => (
  <div className="navbar">
    <h2>Browse our list of recipes </h2>
    <div className="ul">
      <li><a href="/:a">A</a></li>
      <li><a href="/:b">B</a></li>
      <li><a href="/:c">C</a></li>
    </div>
  </div>
);

export default ListByAlphapet;
