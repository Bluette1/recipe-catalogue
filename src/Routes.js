import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import { history } from './store';
import App from './components/App';
import ListByAlphabet from './layouts/ListByAlphabet';
import Details from './components/MealDetails';
import NotFound from './layouts/NotFound';

const Routes = () => (
  <ConnectedRouter history={history}>
    <Switch>
      <Route exact path="/details" component={Details} />
      <Route exact path="/search" component={App} />
      <Route exact path="/" component={ListByAlphabet} />
      <Route component={NotFound} />
    </Switch>
  </ConnectedRouter>
);

export default Routes;
