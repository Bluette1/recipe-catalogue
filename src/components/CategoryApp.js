import '../css/App.css';
import React from 'react';
import MealsList from '../containers/MealsListByCategory';
import AlphabetFilter from './AlphabetFilter';

export default function App() {
  return (
    <div className="content">
      <ul className="mealslist-heading">
        <li>
          Recipe Catalogue
        </li>
        <li><a href="/"><button type="button">Meals</button></a></li>
        <AlphabetFilter />
      </ul>
      <div>
        <MealsList />
      </div>
    </div>
  );
}
