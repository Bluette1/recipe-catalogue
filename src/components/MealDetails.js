import React from 'react';
import PropTypes from 'prop-types';
import QueryString from 'query-string';
import '../css/MealsDetails.css';

const MealDetails = props => {
  const { location: { search } } = props;
  const parsedParams = QueryString.parse(search);
  const {
    img, t, c, i, st, y, ins,
  } = parsedParams;
  return (
    <div>
      <div className="header">
        <ul className="mealslist-heading">
          <li>
            Recipe catalogue
          </li>
          <a href="/"><button type="button">Back</button></a>
        </ul>
      </div>
      {' '}
      <div className="image details-pg">
        <div
          className="image-area details-pg"
          style={{
            backgroundImage: `url(${img})`,
            backgroundRepeat: 'no-repeat',
            backgroundPosition: '0% 0%',
            backgroundSize: 'cover',
          }}
        />
      </div>
      <div className="meal-row details-pg">
        <div className="title-category">
          <p className="category">{c}</p>
          <p className="area category">{st}</p>
          <h4 className="title">{t}</h4>
        </div>
        <div>
          <h4>
            Ingredients:
            {i}
          </h4>
        </div>
      </div>
      <div className="meal-row ins">
        <div>
          <div className="ins-heading">
            <h4 className="instructions">Instructions:</h4>
            <a href={y}><button type="button">Youtube Video</button></a>
          </div>
          <p className="body-ins">{ins}</p>
        </div>
      </div>
    </div>
  );
};

MealDetails.propTypes = {
  location: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default MealDetails;
