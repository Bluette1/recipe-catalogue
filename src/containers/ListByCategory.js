import React, { useState, useEffect } from 'react';
import '../css/ListByAlphabet.css';
import axios from 'axios';

const ListByCategory = () => {
  const [data, setData] = useState([]);

  useEffect(() => {
    axios.get('https://www.themealdb.com/api/json/v1/1/categories.php')
      .then(response => {
        const list = response.data.categories;
        const categoryList = [];
        list.forEach(category => {
          categoryList.push(category.strCategory);
        });
        setData(categoryList);
      });
  }, []);

  return (
    <div className="list-by-alphabet">
      <h2>Browse Available Recipes </h2>
      <div className="ul">
        {data.map(category => (
          <li key={category}><a href={`/filter?c=${category}`}>{category}</a></li>
        ))}
      </div>
    </div>
  );
};

export default ListByCategory;
