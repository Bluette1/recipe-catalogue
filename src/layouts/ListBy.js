import React from 'react';

import '../css/ListByAlphabet.css';

const ListBy = () => (
  <div className="list-by-alphabet">
    <h2>Browse Available Recipes By</h2>
    <div className="ul">
      <li><a href="/alphabet">Alphabet</a></li>
      <li><a href="/category">Category</a></li>
    </div>
  </div>
);

export default ListBy;
