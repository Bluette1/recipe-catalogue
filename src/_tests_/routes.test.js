import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Route } from 'react-router-dom';
import Routes from '../Routes';
import MealDetails from '../components/MealDetails';
import AlphabetApp from '../components/AlphabetApp';
import CategoryApp from '../components/CategoryApp';
import ListByAlphabet from '../layouts/ListByAlphabet';
import NotFound from '../layouts/NotFound';
import ListBy from '../layouts/ListBy';
import ListByCategory from '../containers/ListByCategory';

configure({ adapter: new Adapter() });

let pathMap = {};
describe('routes using array of routers', () => {
  beforeAll(() => {
    const component = shallow(<Routes />);
    pathMap = component.find(Route).reduce((map, route) => {
      const routeMap = map;
      const routeProps = route.props();
      routeMap[routeProps.path] = routeProps.component;
      return routeMap;
    }, {});
  });
  it('should show ListByAlphabet component for /alphabet route', () => {
    const listByAlphabet = pathMap['/alphabet'];
    expect(listByAlphabet).toBe(ListByAlphabet);
  });

  it('should show ListByCategory component for /category route', () => {
    const listByCategory = pathMap['/category'];
    expect(listByCategory).toBe(ListByCategory);
  });

  it('should show CategoryApp component for /filter route', () => {
    const app = pathMap['/filter'];
    expect(app).toBe(CategoryApp);
    expect(app).toMatchSnapshot();
  });

  it('should show AlphabetApp component for /search route', () => {
    const app = pathMap['/search'];
    expect(app).toBe(AlphabetApp);
    expect(app).toMatchSnapshot();
  });

  it('should show ListBy component for / route', () => {
    const homePage = pathMap['/'];
    expect(homePage).toBe(ListBy);
    expect(homePage).toMatchSnapshot();
  });

  it('should show MealDetails component  for /details route', () => {
    const mealDetails = pathMap['/details'];
    expect(mealDetails).toBe(MealDetails);
    expect(mealDetails).toMatchSnapshot();
  });
  it('should show NotFound component for any route that is not defined', () => {
    const unKnownPath = pathMap.undefined;
    expect(unKnownPath).toBe(NotFound);
    expect(unKnownPath).toMatchSnapshot();
  });
});
