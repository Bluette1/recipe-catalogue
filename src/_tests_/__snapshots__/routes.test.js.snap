// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes using array of routers should show App component for /search route 1`] = `[Function]`;

exports[`routes using array of routers should show MealDetails component  for /details route 1`] = `[Function]`;

exports[`routes using array of routers should show NotFound component for any route that is not defined 1`] = `[Function]`;
